packages/world/src/systems/libraries/MoveLib.sol

```diff
@@ -131,17 +47,113 @@ library MoveLib {
     if (playerEntityIdAtCoord.exists()) {
       // If the entity we're moving into is this player, then it's fine as
       // the player will be moved from the old position to the new position
-      bool isSelf = false;
-      for (uint256 i = 0; i < playerEntityIds.length; i++) {
-        if (playerEntityIds[i] == playerEntityIdAtCoord) {
-          isSelf = true;
-          break;
+      require(_isSelf(playerEntityIds, playerEntityIdAtCoord), "Cannot move through a player");
```

Comment:
This feels a bit redundant, can't we just delete the player at the beginning of the move so we know that if there is a player it is not the player being moved?